### PR TITLE
Add workflow option that allows SSH into Google Genomics workers [BA-4966]

### PR DESCRIPTION
**Pull request by acoffman**
This PR adds a workflow specific option (similar to `monitoring_script`) that will spin up an SSH server in a container on Google Genomics workers.

It essentially uses the approach discussed [here](https://github.com/broadinstitute/cromwell/issues/4966#issuecomment-492863979) in an earlier GitHub issue, adding an additional Action that creates a Docker container with Google Genomics Tools'`ssh-server` entrypoint. (See [here]( https://github.com/googlegenomics/pipelines-tools/blob/749315a73e6c3bd5277351e32a365f42198db1ae/pipelines/internal/commands/run/run.go#L759-L766) for Google's implementation.)

I also updated the documentation to include the new parameter, and have tested it in our GCP environment with the attached files.

There is a corresponding JIRA issue here: https://broadworkbench.atlassian.net/browse/BA-4966

Happy to add or fix up anything on here, just let me know.

Thanks!
Adam

`workflow.cwl`
```
#!/usr/bin/env cwl-runner

cwlVersion: v1.0
class: CommandLineTool
baseCommand: sleep
requirements:
  - class: DockerRequirement
    dockerPull: "debian:stretch"
inputs:
  time:
    type: int
    inputBinding:
      position: 1
outputs: []
```

`workflow-inputs.yml`
```
time: 600
```

`options.json`
```
{
  "enable_ssh_access": true
}
```

### CODE REVIEW

**Comment by cjllanwarne** [comment]
Looks good to me on a first pass. 

Just to confirm, did you look into how the pipelines tool implements this (https://github.com/googlegenomics/pipelines-tools#ssh-into-the-worker-machine)? In the back of my mind there's a question over whether there's an existing API switch on the pipelines API for this (vs rolling your own action)

**Comment by acoffman** [comment]
Hi @cjllanwarne, thanks for taking a look!

Yeah the link to "google's implementation" in the [description](https://github.com/googlegenomics/pipelines-tools/blob/749315a73e6c3bd5277351e32a365f42198db1ae/pipelines/internal/commands/run/run.go#L759-L766) is where the Action came from; its essentially a direct port into Scala. You can see [here](https://github.com/googlegenomics/pipelines-tools/blob/749315a73e6c3bd5277351e32a365f42198db1ae/pipelines/internal/commands/run/run.go#L402) where that flag is handled, if its set to true, then the Action is added. 

The `ssh-server` entrypoint is google's own implementation for the pipelines api, found [here](https://github.com/googlegenomics/pipelines-tools/tree/749315a73e6c3bd5277351e32a365f42198db1ae/ssh-server).

As far as I'm aware there's no direct switch on the existing API to have it do that automatically.

**Comment by cjllanwarne** [comment]
Thanks @acoffman, sorry for missing that comment. I've now made an internal clone of your branch to allow our CI tests run over your changes (as #5385).

**Comment by cjllanwarne** [comment]
Looks like the full CI suite is happy with a salted version of commit 832cd0e (see #3858)

**Comment by cjllanwarne** [comment]
Looks like you also need to rebase/merge in `develop` to resolve a conflict in `Google.md`

File: supportedBackends/google/pipelines/v2alpha1/src/main/scala/cromwell/backend/google/pipelines/v2alpha1/api/RemoteAccessAction.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import scala.collection.JavaConverters._
+
+trait RemoteAccessAction {
+
+  def remoteAccessActions(createPipelineParameters: CreatePipelineParameters, mounts: List[Mount]) : List[Action] = {
+    val workflowOptions = createPipelineParameters.jobDescriptor.workflowDescriptor.workflowOptions
+
+    workflowOptions.getBoolean(WorkflowOptionKeys.EnableRemoteAccess).toOption match {
+      case Some(enable) if enable => remoteAccessAction(mounts)
```
[the remark this answers is not in the document]
Comment (acoffman) [reply supportedBackends/google/pipelines/v2alpha1/src/main/scala/cromwell/backend/google/pipelines/v2alpha1/api/RemoteAccessAction.scala]: Good suggestion, thanks!

File: docs/wf_options/Google.md [hunk 2]
```diff
@@ -11,6 +11,7 @@ Keys | Possible Values | Description
 `auth_bucket` |`string` |     A GCS URL that only Cromwell can write to.  The Cromwell account is determined by the `google.authScheme` (and the corresponding `google.userAuth` and `google.serviceAuth`). Defaults to the the value in [jes_gcs_root](#jes_gcs_root).
 `monitoring_script` |`string` |   Specifies a GCS URL to a script that will be invoked prior to the user command being run.  For example, if the value for monitoring_script is `"gs://bucket/script.sh"`, it will be invoked as `./script.sh > monitoring.log &`.  The value `monitoring.log` file will be automatically de-localized.
 `monitoring_image` |`string` |   Specifies a Docker image to monitor the task. This image will run concurrently with the task container, and provides an alternative mechanism to `monitoring_script` (the latter runs *inside* the task container). For example, one can use `quay.io/broadinstitute/cromwell-monitor`, which reports cpu/memory/disk utilization metrics to [Stackdriver](https://cloud.google.com/monitoring/).
+`enable_remote_access` |`boolean` | If set to true, will enable SSH access to the Google Genomics worker machines
```
[the remark this answers is not in the document]
Comment (acoffman) [reply docs/wf_options/Google.md]: Done.

File: docs/wf_options/Google.md [hunk 3]
```diff
@@ -19,10 +20,11 @@ Keys | Possible Values | Description
   "jes_gcs_root": "gs://my-bucket/workflows",
   "google_project": "my_google_project",
   "refresh_token": "1/Fjf8gfJr5fdfNf9dk26fdn23FDm4x",
-  "google_compute_service_account": " my-new-svcacct@my-google-project.iam.gserviceaccount.com"
+  "google_compute_service_account": " my-new-svcacct@my-google-project.iam.gserviceaccount.com",
   "auth_bucket": "gs://my-auth-bucket/private",
   "monitoring_script": "gs://bucket/script.sh",
   "monitoring_image": "quay.io/broadinstitute/cromwell-monitor",
+  "enable_remote_access": false,
```
[the remark this answers is not in the document]
Comment (acoffman) [reply docs/wf_options/Google.md]: Done.

**Comment by acoffman** [comment]
Thanks for the review guys! I believe I've addressed your comments, let me know if there's anything further you need me to do.

**Comment by acoffman** [comment]
Looking at other open PRs, it occurs to me that this change will perhaps become outdated with the completion of #5386 

Would it make the most sense for me to make a new PR once its merged to incorporate it into the v2beta backend or for you guys to hold off merging this PR and having me add to it once  #5386 is completed?

**Comment by cjllanwarne** [comment]
@grsterin do you have an opinion on the comment above?

**Comment by grsterin** [comment]
@acoffman, @cjllanwarne, I think it would be better to wait for #5386 to be merged and then add this feature into both v2alpha1 and v2beta APIs at once. It'll take just some copy-pasting.

**Comment by acoffman** [comment]
Sounds good - I'll wait for that to be merged and then append to this PR!

**Comment by grsterin** [comment]
@acoffman #5386 has been merged to develop, so you may go on with this PR and copy your changes into v2beta package. Thanks for waiting!

**Comment by acoffman** [comment]
@grsterin added support for the updated backend and tested with the same workflow/inputs as above, but using the new backend and all seems to be well. Thanks for taking a look at this!

**Comment by cjllanwarne** [comment]
Thanks again @acoffman - I just repushed the CI clone branch (PR #5385) with your latest changes so we should be able to merge as soon as that passes.

**Comment by cjllanwarne** [comment]
The tests in #5385 have passed despite codecov's unhappiness on this PR, and with two thumbs I think this is ready for merging. Thanks again @acoffman !

**Pull request merged**